gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard-vxlan/checkuplink

```diff
@@ -0,0 +1,75 @@
+#!/bin/sh
+
+xor2() {
+	echo -n "${1:0:1}"
+	echo -n "${1:1:1}" | tr '0123456789abcdef' '23016745ab89efcd'
+}
+
+interface_linklocal() {
+	# We generate a predictable v6 address
+	local macaddr="$(echo $(uci get wireguard.wireguard.privatekey | wg pubkey) |md5sum|sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/')"
+	local oldIFS="$IFS"; IFS=':'; set -- $macaddr; IFS="$oldIFS"
+	echo "fe80::$(xor2 "$1")$2:$3ff:fe$4:$5$6"
+}
+
+# Do we already have a private-key? If not generate one
+temp=$(uci get wireguard.wireguard.privatekey);
+if [ "$?" -ne "0" ]; then
+	uci set wireguard.wireguard.privatekey=$(wg genkey);
+	uci commit wireguard
+fi
+
+# Is wireguard enabled?
+if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wireguard.wireguard.enabled)" == "1" ]; then
+
+	#We assume we are not connected by default
+	CONNECTED=0
+
+	MESH_VPN_IFACE=$(uci get wireguard.wireguard.iface)
+
+	# Check connectivity to supernode
+	ping -c1 -w2  $(wg  | grep fe80 | cut -d'/' -f1 | awk '{print $3}')%$MESH_VPN_IFACE
+	if [ $? == 0 ]; then
+			CONNECTED=1
+	fi
+
+	# If we don't have a connection we try to connect
+	if [ $CONNECTED != 1 ]; then                                    
+
+		# Get the number of configured peers and randomly select one
+		NUMBER_OF_PEERS=$(uci -q show wireguard | egrep -ce peer_[0-9]+.endpoint)
+		PEER=$(awk 'BEGIN{srand();print int(rand()*($NUMBER_OF_PEERS)+1) }')
+
+		# Delete Interfaces
+		ip link set nomaster bat0 dev vx_mesh_vpn &> /dev/null
+		ip link delete dev vx_mesh_vpn &> /dev/null
+		ip link del $MESH_VPN_IFACE &> /dev/null
+		PUBLICKEY=$(uci get wireguard.wireguard.privatekey | wg pubkey)
+		echo $(uci get wireguard.wireguard.privatekey) > /tmp/wgpriv
+
+		# For future use
+		gluon-wan wget -q http://gw03.ext.ffmuc.net:5000/wg-public-key/$PUBLICKEY
+
+		# Bring up the wireguard interface
+		ip link add dev $MESH_VPN_IFACE type wireguard
+		wg set $MESH_VPN_IFACE private-key /tmp/wgpriv
```

Comment:
If this is executing in bash, you could do `wg set $MESH_VPN_IFACE private-key <(uci get wireguard.wireguard.privatekey)`. If you're not using bash, you can do `uci get wireguard.wireguard.privatekey | wg set wg0 private-key /dev/stdin` or `uci get wireguard.wireguard.privatekey | wg set wg0 private-key /proc/self/fd/0`.

Reply:
Executes in busybox ash.

Reply:
Alright, so` uci get wireguard.wireguard.privatekey | wg set wg0 private-key /dev/stdin` or `uci get wireguard.wireguard.privatekey | wg set wg0 private-key /proc/self/fd/0`.